detectable/src/main/java/com/synopsys/integration/detectable/detectables/projectinspector/ProjectInspectorParser.java

```diff
@@ -72,20 +75,32 @@ public List<CodeLocation> processModules(JsonReader reader) throws IOException {
             if (moduleId != null) {
                 JsonObject module = JsonParser.parseReader(reader).getAsJsonObject();
                 ProjectInspectorModule projectInspectorModule = gson.fromJson(module, ProjectInspectorModule.class);
-                CodeLocation codeLocation = codeLocationFromModule(projectInspectorModule);
-                codeLocations.add(codeLocation);
+                if(projectInspectorModule.components != null) {
+                    if (includeShadedDependencies) {
+                        processShadedDependencies(projectInspectorModule);
+                    } else {
+                        CodeLocation codeLocation = codeLocationFromModule(projectInspectorModule);
+                        codeLocations.add(codeLocation);
+                    }
+                } else {
+                    versionMismatch = true;
+                }
             }
         }
         reader.endObject();
 
+        if(versionMismatch) {
+            logger.info("From Detect 9.5.0 and onwards, we will support the latest version of Project Inspector i.e. 2024.2.0. If you are using the detect.project.inspector.path property, please use the latest version mentioned.");
```

Comment:
PI might have multiple modules in the output file, so the above line was getting logged multiple times while parsing the execution file, so made sure that it gets printed only once. It is unlikely to happen the components are null in one module and not other as PI will not send empty modules in the output file.

Reply:
similar to earlier comment I would not think we'd want to recommend the example project inspector version